src/main/java/seedu/recipe/logic/parser/EditStepCommandParser.java

```diff
@@ -37,7 +37,7 @@ public EditStepCommand parse(String args) throws ParseException {
             index = ParserUtil.parseIndex(argsArray[0]);
             stepNumber = ParserUtil.parseIndex(argsArray[1]).getZeroBased();
         } catch (ParseException pe) {
-            throw new ParseException(ParserUtil.MESSAGE_INVALID_INDEX);
+            throw new ParseException(pe.getMessage());
```

Comment:
Im not too sure whether we are able to do this, but could we not catch this ParseException, since we are catching and throwing a parseException with the same message?

So if a ParseException was thrown in this try block, it would just be thrown back to whoever called this method to be handled, and this catch block here would only catch the ArrayIndexOutOfBoundsException and throw a ParseException?

Reply:
yes you have a good point. was an oversight on my part. thank you for pointing it out! I have removed this unnecessary catch block in my latest commit :)